src/flows/shared/components/WizardFooter/WizardFooter.tsx

```diff
@@ -65,18 +70,60 @@ function WizardFooter({
     next();
   }, [back, collectionId, next, onNext]);
 
+  const { showModal } = useModalActions();
+
   const handlePreviousClick = useCallback(() => {
```

Comment:
On click, we show a `GenericActionModal` which runs the specific action and then hides the modal afterwards. These behaviors are in line with what we talked about in Linear